home-assistant-voice.yaml

```
@@ -703,9 +703,9 @@ light:
             }
       - addressable_lambda:
           name: "Center Button Touched"
-          update_interval: 16ms
+          update_interval: 1ms
           lambda: |-
-            auto light_color = id(led_ring).current_values;
+            auto light_color = id(center_button).state ? id(voice_assistant_leds).current_values : id(led_ring).current_values;
```

I am not sure the `voice_assistant_leds` has a proper color...
Since almost everything on this `light` is using effects that drive the color ...

For example, `on_error` the `voice_assistant_leds` becomes red. 
If you enter this right after `on_error`, pretty sure the `voice_assistant_leds` would be red...

---

Ooooooouh ! I get what you are doing. 
It's super smart.

Here is a proposal that (I think) is cleaner.
Tell me what you think.

ON the effect itself (Here), you can use the `initial_run` bool to capture the `led_ring` color.
Then on every subsequent run, you capture the `voice_assistant_leds`

Since now you are now changing the `voice_assistant_leds` and not the `led_ring` during dial rotation, it would be super clean.

I would prefer this, than this test.

The test is weird. 
`id(center_button).state` is always `on` on this effect. The effect is activated when the button is pressed!

---

```suggestion
            light::LightColorValues light_color = LightColorValues()
            if (initial_run) {
              light_color = id(led_ring).current_values;
            } else {
              light_color = id(voice_assistant_leds).current_values;
            }
```

Something like this?

---

So my question is this: when/where is this effect de-activated? 😇
I only found one spot where the effect is set back to `none`.

---

Put another way: you need this test because if/when the effect is left engaged, then `led_ring` might not get set to the correct color. What happens if the user changes the color from HA and this effect is still engaged?

---

Check the main script control_leds 
It's a master script setting all the effects on the LED from a central place. 

When the button is pressed, this effect is applied, when it is depressed it's "whatever has more importance at that moment". 

It can go back to the "replying effect" for example if you press the button while the VA replies

---

> What happens if the user changes the color from HA and this effect is still engaged?

I think you can safely assume this is out of scope. 
You need 3 hands 😅
One that press, one that rotates, one that control HA. 

Honneslty if it's the only issue, we have no issue 🥹

---

So the reason for the test is to account for when the effect is engaged but the button is not pressed.

The effect may not be disengaged until something else changes it. In other words, releasing the button does not specifically disengage this effect, so the effect needs to be able to account for when it's engaged but the button is not pressed. It has nothing to do with the initial run of the effect.

I just checked and it happens to work without the test as it is right now, but only because something else sets the effect back to "Volume Display" when the button is released. BUT, _if_ we can remove it, unexpected behavior _will occur_ if the effect is ever left engaged while the button is not pressed.

FWIW, I really don't think we should remove it. 🤷🏻

---

Then I trust you ! No pb. 

I still wonder about the "catching the current color of the internal voice assistant LEDs ..."

They often have no color, just effects set ... 

I'm a bit afraid of that. 

Let me try to find a case where it could go wrong.

---

Ah ! Let's try this ! Easy 

- Set the LED Ring in HA to Green. 
Turn it off 
- Disable the config entry on HA 
- The device should loose its connection and display a red twinkle effect. Note this time it's really setting the color of the voice assistant light to red. 
- Re-enable the config entry 
- Long press the button (or just press it ☺️)

What color do you see ? Red or green? 

It should be green. It's the user choice, but I am afraid it's red

---

Did those steps exactly and, at the final step, the LEDs came on green. 😄

---

What the ... how ? 🤣😅😂

---

Ah, after the long press, the LEDs are red.
But if I do not do the long press, they are green.

---

Sorry, had to edit my comments as I was playing with it more. haha

---

Exactly! Yes exactly. They are red cause the VA light is still "red"

---

I see what's happening. I think we just need an additional condition to check whether the color is being adjusted.

---

Thx! And sorry for not being next to the device it's more complex for me to test things

---

Ok, I have fixed this, now the `light` behavior is exactly as expected based on your test above. 😄🎉